packages/react-instantsearch/src/core/createInstantSearchManager.test.js

```diff
@@ -24,15 +24,16 @@ const response = {
   ],
 };
 client.search = jest.fn((queries, cb) => {
+  const clonedResponse = JSON.parse(JSON.stringify(response));
```

Comment:
Yep the Helper assumes that the response comes from a source that will always return new instances. :( Sorry for the work that it created to you :(